src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -42,12 +42,25 @@ public class DurableOrchestrationClient
         /// </summary>
         /// <param name="request">The HTTP request that triggered the current function.</param>
         /// <param name="instanceId">The unique ID of the instance to check.</param>
-        /// <returns>An HTTP response which may include a 202 and locaton header.</returns>
+        /// <returns>An HTTP response which may include a 202 and location header.</returns>
         public HttpResponseMessage CreateCheckStatusResponse(HttpRequestMessage request, string instanceId)
         {
             return this.config.CreateCheckStatusResponse(request, instanceId, this.attribute);
         }
 
+        /// <summary>
+        /// Creates an HTTP response for checking the status of the specified instance supporting synchronous response as well. 
+        /// </summary>
+        /// <param name="request">The HTTP request that triggered the current function.</param>
+        /// <param name="instanceId">The unique ID of the instance to check.</param>
+        /// <param name="timeout">Total allowed timeout for output from the durable function.</param>
+        /// <param name="retryInterval">Timeout between checks for output from the durable function. The default value is 5 seconds.</param>
+        /// <returns>An HTTP response which may include a 202 and location header or a 200 with the durable function output in the response body.</returns>
+        public async Task<HttpResponseMessage> CreateCheckStatusResponse(HttpRequestMessage request, string instanceId, TimeSpan timeout, TimeSpan? retryInterval)
```

Comment:
It just occurred to me from reading the samples that this method name can be misleading. I suggest an alternative name that better describes the behavior, like `WaitForCompletionOrCreateCheckStatusResponseAsync`. The important behavior to communicate in the method name is that the method call may not complete immediately (unlike `CreateCheckStatusResponse`, which always completes immediately). I would suggest making that name change both in DurableOrchestrationClient and in DurableTaskExtension.

Reply:
I like when the methods and variables are self-descriptive and the dev does not need to go and study the documentation in details. I will work on that! Thank you for the great feedback!